django_snowflake/features.py

```diff
@@ -79,6 +78,63 @@ class DatabaseFeatures(BaseDatabaseFeatures):
         'expressions.tests.FTimeDeltaTests.test_date_subtraction',
         'expressions.tests.FTimeDeltaTests.test_datetime_subtraction',
         'expressions.tests.FTimeDeltaTests.test_time_subtraction',
+        # https://github.com/django/django/commit/7deeabc7c7526786df6894429ce89a9c4b614086
+        # When trying to pass the json_path as a query parameter, Snowflake
+        # Python Connector's Cursor._preprocess_pyformat_query() interpolates
+        # (command % processed_params) and adds incorrect quotes, e.g.
+        # "MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE"':%total' instead of
+        # "MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE":%total.
+        'model_fields.test_jsonfield.TestQuerying.test_key_escape',
+        'model_fields.test_jsonfield.TestQuerying.test_key_sql_injection_escape',
+        # https://github.com/django/django/commit/a88fab1bca33c037bd1a12459c215d7bc1247735
+        # Integer looks like "MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE"[123] need
+        # to be rewritten as "MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE":"123"
+        # But this breaks nested cases like:
+        # IS_NULL_VALUE("MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE":d[1].f)
+        # which would be rewritten as:
+        # IS_NULL_VALUE("MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE":"d"."1"."f")
+        # which is semantically different (treating 1 as key vs. array).
+        # Could possibly use "OR" to check both cases but the number of ORs
+        # need increases exponentially for nested lookups.
+        'model_fields.test_jsonfield.TestQuerying.test_has_key_number',
+        # null issue:
```

Comment:
@sfc-gh-mkeller I'm wondering if have you a sense of whether or not this issue and the below issues mentioning "needs to operate as" (where `PARSE_JSON()` is needed on the right-hand side of the query) will be solved by https://github.com/snowflakedb/snowflake-connector-python/issues/244? I don't know if it's possible to work around here until then.

Also, feel free to comment on any of the other failures here if you have any suggestions.

Reply:
I think it'd make sense if you did `"MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE"[123]` if the key is an integer and do the `"MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE":"123"` if the key is a string.
The JSON standard only allows string keys.

Reply:
I think your comment refers to `test_has_key_number'` which I solved last night in the latest commit. I was actually asking about cases like this:
```
WHERE TO_JSON("MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE":k) = '{"l": "m"}'
```
where the right-hand side parameter is treated as a string literal rather than as JSON.

A working version of this query is:
```
WHERE TO_JSON("MODEL_FIELDS_NULLABLEJSONMODEL"."VALUE":k) = PARSE_JSON('{"l": "m"}')
```
but I'm not sure this is feasible to implement here. I thought maybe when/if the Python connector gains awareness of VARIANT type, the first query might work as expected.

Reply:
I've documented this as a known limitation: https://github.com/cedar-team/django-snowflake/issues/58.